x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -2895,6 +2921,216 @@ public void testNestedMultiExpressionsInGroupingAndAggs() {
     }
 
+    /**
+     * Expects
+     * Project[[x{r}#5]]
+     * \_Eval[[____x_AVG@9efc3cf3_SUM@daf9f221{r}#18 / ____x_AVG@9efc3cf3_COUNT@53cd08ed{r}#19 AS __x_AVG@9efc3cf3, __x_AVG@
+     * 9efc3cf3{r}#16 / 2[INTEGER] + __x_MAX@475d0e4d{r}#17 AS x]]
+     *   \_Limit[500[INTEGER]]
+     *     \_Aggregate[[],[SUM(salary{f}#11) AS ____x_AVG@9efc3cf3_SUM@daf9f221, COUNT(salary{f}#11) AS ____x_AVG@9efc3cf3_COUNT@53cd0
+     * 8ed, MAX(salary{f}#11) AS __x_MAX@475d0e4d]]
+     *       \_EsRelation[test][_meta_field{f}#12, emp_no{f}#6, first_name{f}#7, ge..]
+     */
+    public void testStatsExpOverAggs() {
+        var plan = optimizedPlan("""
+            from test
+            | stats x = avg(salary) /2 + max(salary)
+            """);
+
+        var project = as(plan, Project.class);
+        assertThat(Expressions.names(project.projections()), contains("x"));
+        var eval = as(project.child(), Eval.class);
+        var fields = eval.fields();
+        assertThat(Expressions.name(fields.get(1)), is("x"));
+        // sum/count to compute avg
+        var div = as(fields.get(0).child(), Div.class);
+        // avg + max
+        var add = as(fields.get(1).child(), Add.class);
+        var limit = as(eval.child(), Limit.class);
+        var agg = as(limit.child(), Aggregate.class);
+        var aggs = agg.aggregates();
+        assertThat(aggs, hasSize(3));
+        var sum = as(Alias.unwrap(aggs.get(0)), Sum.class);
+        assertThat(Expressions.name(sum.field()), is("salary"));
+        var count = as(Alias.unwrap(aggs.get(1)), Count.class);
+        assertThat(Expressions.name(count.field()), is("salary"));
+        var max = as(Alias.unwrap(aggs.get(2)), Max.class);
+        assertThat(Expressions.name(max.field()), is("salary"));
+    }
+
+    /**
+     * Expects
+     * Project[[x{r}#1915, y{r}#1919, z{r}#1922]]
+     * \_Eval[[____x_AVG@e03a7a5c_SUM@acf776e3{r}#1939 / ____x_AVG@e03a7a5c_COUNT@39975522{r}#1940 AS __x_AVG@e03a7a5c,
+     *         __x_AVG@e03a7a5c{r}#1933 + __x_MAX@4e09cbd5{r}#1934 AS x,
+     *         __y_MIN@80cee21c{r}#1935 + 10[INTEGER] - __y_MEDIAN@705fccec{r}#1936 AS y,
+     *        languages{f}#1926 % 2[INTEGER] AS z]]
+     *   \_Limit[500[INTEGER]]
+     *     \_Aggregate[[z{r}#1922],[SUM(____x_AVG@e03a7a5c_AVG@e03a7a5c{r}#1937) AS ____x_AVG@e03a7a5c_SUM@acf776e3,
+     *     COUNT(____x_AVG@e03a7a5c_AVG@e03a7a5c{r}#1937) AS ____x_AVG@e03a7a5c_COUNT@39975522,
+     *     MAX(emp_no{f}#1923) AS __x_MAX@4e09cbd5,
+     *     MIN(____y_MIN@80cee21c_MIN@80cee21c{r}#1938) AS __y_MIN@80cee21c,
+     *     PERCENTILE(salary{f}#1928,50[INTEGER]) AS __y_MEDIAN@705fccec]]
+     *       \_Eval[[languages{f}#1926 % 2[INTEGER] AS z,
+     *               salary{f}#1928 % 3[INTEGER] AS ____x_AVG@e03a7a5c_AVG@e03a7a5c,
+     *               emp_no{f}#1923 / 3[INTEGER] AS ____y_MIN@80cee21c_MIN@80cee21c]]
```

Comment:
nit: the attribute names are constructed from the aggs where they will be used, but not from the expression where they will be used; this makes it a bit hard to read this bottom-up (and top-down is also hard). I think it will also lead to tricky attribute names in cases like `y = min(emp_no /3) - min(emp_no + 2)`, as both attributes will be called `____y_MIN@...._MIN@....`.
Maybe this would be more consistent?
```suggestion
     *               emp_no{f}#1923 / 3[INTEGER] AS y_SUB1_MIN]]
```
(resp. `...SUB0...` for the left hand side)